image_proc/CMakeLists.txt

```diff
@@ -49,20 +56,30 @@ target_compile_definitions(debayer
   PRIVATE "COMPOSITION_BUILDING_DLL")
 
 ament_target_dependencies(debayer
-  "sensor_msgs"
-  "cv_bridge"
-  "class_loader"
-  "image_transport"
-  "rclcpp"
-  "rclcpp_components"
-  )
+  ${dependencies}
+)
 
 rclcpp_components_register_nodes(debayer "image_proc::DebayerNode")
 set(node_plugins "${node_plugins}image_proc::DebayerNode;$<TARGET_FILE:debayer>\n")
 
+add_library(resize SHARED
+  src/resize.cpp
+  )
+
+target_compile_definitions(resize
+  PRIVATE "COMPOSITION_BUILDING_DLL")
+
+ament_target_dependencies(resize
+  ${dependencies}
+)
+
+rclcpp_components_register_nodes(resize "image_proc::ResizeNode")
+set(node_plugins "${node_plugins}image_proc::ResizeNode;$<TARGET_FILE:resize>\n")
```

Comment:
To follow recommended best-practices for exporting components as shared libraries, the following should be added (see [this post](https://discourse.ros.org/t/ament-best-practice-for-sharing-libraries/3602) for more details):

```
ament_export_interfaces(
  export_resize HAS_LIBRARY_TARGET
  export_debayer HAS_LIBRARY_TARGET
  export_rectify HAS_LIBRARY_TARGET
)
ament_export_libraries(
  resize
  debayer
  rectify
)
```

And this would replace the `install` command below:

```
install(
  TARGETS resize
  EXPORT export_resize
  ARCHIVE DESTINATION lib
  LIBRARY DESTINATION lib
  RUNTIME DESTINATION bin
)
install(
  TARGETS debayer
  EXPORT export_debayer
  ARCHIVE DESTINATION lib
  LIBRARY DESTINATION lib
  RUNTIME DESTINATION bin
)
install(
  TARGETS rectify
  EXPORT export_rectify
  ARCHIVE DESTINATION lib
  LIBRARY DESTINATION lib
  RUNTIME DESTINATION bin
)
```

The advantage of exporting the targets/interfaces as above instead of how it's being done now is that downstream packages can link against the namespaced interface (e.g. `image_proc::ResizeNode`) instead of the library name (e.g. `resize`). It's also the recommended way going forward.

Reply:
Awesome thanks, hadn't seen that approach before.